File: Mar-18/15-03-18/java/02-Modified-Sequence.java
import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int diff = input.nextInt();
        
        for(int i=1;i<n;i++)
        {
            int tmp = input.nextInt();
            diff = tmp-diff;
            if(diff<0)
                break;
        }
        if(diff==0)
            System.out.println("YES");
        else
            System.out.println("NO");
        
        
    }
}

File: Mar-18/15-03-18/c++/02-Modified-Sequence.cpp
#include<bits/stdc++.h>
 
using namespace std;
 
int main()
{
    int n;
    cin>>n;
    int ar[n];
    for(int i=0;i<n;i++)
        cin>>ar[i];
    
    int diff=ar[0];
    for(int i=1;i<n;i++)
    {
        diff=ar[i]-diff;
        if(diff<0)
            break;
    }
    if(diff==0)
        cout<<"YES\n";
    else
        cout<<"NO\n";
    
}
